src/components/NewMessageForm/AudioRecorder/AudioRecorder.vue

```diff
@@ -0,0 +1,350 @@
+				} else {
+					showError(t('spreed', 'Microphone either not available or disabled in settings'))
+				}
+				return
+			}
+
+			// Create a mediarecorder to capture the stream
+			try {
+				this.mediaRecorder = new MediaRecorder(this.audioStream)
+			} catch (exception) {
+				console.debug(exception)
+				this.audioStream.getTracks().forEach(track => track.stop())
+				this.audioStream = null
+				showError(t('spreed', 'Error while recording audio'))
+				return
+			}
+
+			// Add event handler to onstop
+			this.mediaRecorder.onstop = this.generateFile
+
+			// Add event handler to ondataAvailable
+			this.mediaRecorder.ondataavailable = (e) => {
+				this.chunks.push(e.data)
+			}
+
+			try {
+				// Start the recording
+				this.mediaRecorder.start()
+			} catch (exception) {
+				console.debug(exception)
+				this.aborted = true
+				this.stop()
+				this.resetComponentData()
+				showError(t('spreed', 'Error while recording audio'))
+				return
+			}
+
+			console.debug(this.mediaRecorder.state)
+
+			// Start the timer
+			this.recordTimer = setInterval(() => {
+				if (this.recordTime.seconds === 59) {
+					this.recordTime.minutes++
+					this.recordTime.seconds = 0
+				}
+				this.recordTime.seconds++
+			}, 1000)
+			// Forward an event to let the parent NewMessageForm component
+			// that there's an undergoing recording operation
+			this.$emit('recording', true)
+		},
+
+		/**
+		 * Stop the mediaRecorder
+		 */
+		stop() {
+			this.mediaRecorder.stop()
+			clearInterval(this.recordTimer)
+			this.$emit('recording', false)
+		},
+
+		/**
+		 * Generate the file
+		 */
+		generateFile() {
+			this.audioStream.getTracks().forEach(track => track.stop())
+			if (!this.aborted) {
+				this.blob = new Blob(this.chunks, { 'type': 'audio/mpeg-3' })
+				// Generate file name
+				const fileName = this.generateFileName()
+				// Convert blob to file
+				const audioFile = new File([this.blob], fileName)
+				this.$emit('audioFile', audioFile)
+				this.$emit('recording', false)
+				this.URL = window.URL.createObjectURL(this.blob)
+			}
+			this.resetComponentData()
+		},
+
+		/**
+		 * Aborts the recording operation.
+		 */
+		abortRecording() {
+			this.aborted = true
+			this.stop()
+		},
+
+		/**
+		 * Resets this component to its initial state
+		 */
+		resetComponentData() {
+			this.audioStream = null
+			this.mediaRecorder = null
+			this.chunks = []
+			this.blob = null
+			this.aborted = false
+			this.isAudiorecorderActive = false
+			this.recordTime = {
+				minutes: 0,
+				seconds: 0,
+			}
+		},
+
+		generateFileName() {
+			const token = this.$store.getters.getToken()
+			const conversationName = this.$store.getters.conversation(token).name
+			const today = new Date()
+			const time = today.getFullYear() + '-' + today.getMonth() + '-' + today.getDay() + ' at ' + today.getHours() + ':' + today.getMinutes() + ':' + today.getSeconds()
+			return `${t('spreed', 'Talk recording from')} ${time} (${conversationName}).mp3`
```

Comment:
* Let's add leading zeros to month, day, hour, minute, second (https://stackoverflow.com/a/3605248)
* Colon is an issue on windows and would break the sync client => using `-` like ubuntu screenshots
* Got rid of untranslated `at` between date and time replaced with a space like ubuntu screenshots
* Made the translation proper so it uses placeholders.

```suggestion
			const conversation = this.$store.getters.conversation(token).name
			const today = new Date()
			let time = today.getFullYear() + '-' + ('0' + today.getMonth()).slice(-2) + '-' + ('0' + today.getDay()).slice(-2)
			time += ' ' + ('0' + today.getHours()).slice(-2) + '-' + ('0' + today.getMinutes()).slice(-2) + '-' + ('0' + today.getSeconds()).slice(-2)
			return t('spreed', 'Talk recording from {time} ({conversation})', { time, conversation }) + ' .mp3'
```

Reply:
better use the moment library for proper date formatting ? I believe we do so already for image uploads

Reply:
ref https://github.com/nextcloud/spreed/blob/5249c1390a04968eefc8a9ee183306d646be7c53/src/store/fileUploadStore.js#L206

Reply:
Moment is deprecated, but sure